File: Convert_EMBL-to-GFF_24-08-17.py
#!usr/usr/bin/python

###################
# Import packages #
###################

import sys
from BCBio import GFF
from Bio import SeqIO

# Python script to convert from EMBL to GFF format

# Command line structure:
# python Convert_EMBL-to-GFF.py input.embl output.gff

# Code taken from to-gff tool. Available here: https://github.com/mscook/to-gff
# Note that if converting EMBL file from RATT then you'll need to substitute the first line from the original EMBL file

###############
# Check input #
###############

if len(sys.argv) != 3:
	print "Python script to convert EMBL formatted file into GFF formatted file"
	print "Author: Joseph Crispell"
	print "Created: 24-08-17\n"
	print "Command line structure:"
	print "\tpython Convert_EMBL-to-GFF.py input.embl output.gff"
	print "\t\tinput.embl\tPath to input file in EMBL format"
	print "\t\toutput.gff\tPath to output file (will be in GFF format)"
	sys.exit()

##################################
# Get the input and output files #
##################################

input = open(sys.argv[1])
output = open(sys.argv[2], 'w')

#########################################
# Convert input EMBL to output GFF file #
#########################################

GFF.write(SeqIO.parse(input, "embl"), output)

File: Convert_EMBL-to-GFF_24-08-17.pl
#!/usr/local/bin/perl -w
use strict;
use Bio::SeqIO;

# Script to convert EMBL format to GFF format
# Command line structure:
# perl Convert_EMBL-to-GFF.pl file.embl > output.gff

## Script edited from: http://ratt.sourceforge.net/transform.html

# Check that input arguements were given correctly
if (scalar(@ARGV) != 1) {
    die "USAGE: Convert_EMBL-to-GFF.pl file.embl > output.gff\n"; 
}

###########################################
# Convert input file into gff output file #
###########################################

# Note that using bioperl functions

# Initialise variables for creating the output file
my $line;
my @cols;

# Print header

# Read in the input file
my $embl = shift @ARGV;
my $input = Bio::SeqIO->new(-file=>$embl,-format=>'EMBL');
while (my $seq = $input->next_seq) {
  for my $feature ($seq->top_SeqFeatures) {
  
	# Get the new line
	$line = $feature->gff_string;
	
	# Split the line into its columns
	@cols = split /\t/, $line;
	
	# Replace the second column with "feature"
	$cols[1] = "feature";
	
	# Remake the line
	$line = join("\t", @cols);
	
	print "$line\n";
  }
}